File: 0000-0999/0120.Triangle.py
class Solution(object):
    def minimumTotal(self, triangle):
        """
        :type triangle: List[List[int]]
        :rtype: int
        """
        if not triangle:
            return 0
        '''
        for row in xrange(len(triangle)-2, -1, -1):
            for col in xrange(len(triangle[row])):
                triangle[row][col] += min(triangle[row+1][col], triangle[row+1][col+1])
        return triangle[0][0]
        '''
        temp = [0] * (len(triangle)+1)
        for row in xrange(len(triangle)-1, -1, -1):
            for col in xrange(len(triangle[row])):
                temp[col] = triangle[row][col] + min(temp[col], temp[col+1])
        return temp[0]
        

File: 0000-0999/0120.Triangle.go
func minimumTotal(triangle [][]int) int {
    n := len(triangle)
    dp := make([]int, n)
    for j := 0; j < n; j++ {
        dp[j] = triangle[n-1][j]
    }
    for i := n-2; i >= 0; i-- {
        for j := 0; j <= i; j++ {
            dp[j] = triangle[i][j] + min(dp[j], dp[j+1])
        }
    }
    return dp[0]
}